server/endpoint/endpoint.go

```diff
@@ -16,9 +19,35 @@ type Config struct {
 
 // New creates a new endpoint with given configuration.
 func New(config Config) (*Endpoint, error) {
-	return &Endpoint{}, nil
+	var err error
+
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	if config.Service == nil || config.Service.Healthz == nil || config.Service.Healthz.K8s == nil {
+		return nil, microerror.Maskf(invalidConfigError,
+			"config.Service or it's Healthz descendents must not be empty")
+	}
```

Comment:
I'd say `Service` created with `New` should be validated, so it's enough to check if it isn't `nil`.